client/src/features/sessionsV2/StartSessionButton.tsx

```diff
@@ -48,15 +50,34 @@ export default function StartSessionButton({
       slug,
     }
   );
+  const environment = launcher?.environment;
+  const isExternalImageEnvironment =
+    environment?.environment_kind === "CUSTOM" &&
+    environment?.environment_image_source === "image";
+  const { data, isLoading } = useGetSessionsImagesQuery(
+    environment &&
+      environment.environment_kind === "CUSTOM" &&
+      environment.container_image
+      ? { imageUrl: environment.container_image }
+      : skipToken
+  );
+
+  const force = isExternalImageEnvironment && !isLoading && !data?.accessible;
```

Comment:
nitpick: I think the variable name can be misleading here.

I would do:

```suggestion
  const imageNotAccessible = isExternalImageEnvironment && !isLoading && !data?.accessible;
```